presto-main/src/test/java/com/facebook/presto/cost/TestComparisonStatsCalculator.java

```diff
@@ -571,46 +597,99 @@ public void symbolToSymbolEqualStats()
                             .distinctValuesCount(10)
                             .nullsFraction(0);
                 })
-                .symbolStats("z", zUnchanged.apply(rowCount));
+                .symbolStats("z", equalTo(capNDV(zStats, rowCount)));
 
         // Partially overlapping ranges
         rowCount = 8.4375;
         assertCalculate(new ComparisonExpression(EQUAL, new SymbolReference("x"), new SymbolReference("w")))
                 .outputRowsCount(rowCount)
                 .symbolStats("x", symbolAssert -> {
-                    symbolAssert.averageRowSize(4)
+                    symbolAssert.averageRowSize(6)
                             .lowValue(0)
                             .highValue(10)
                             .distinctValuesCount(15)
                             .nullsFraction(0);
                 })
                 .symbolStats("w", symbolAssert -> {
-                    symbolAssert.averageRowSize(8)
+                    symbolAssert.averageRowSize(6)
                             .lowValue(0)
                             .highValue(10)
                             .distinctValuesCount(15)
                             .nullsFraction(0);
                 })
-                .symbolStats("z", zUnchanged.apply(rowCount));
+                .symbolStats("z", equalTo(capNDV(zStats, rowCount)));
 
         // None of the ranges is included in the other, and one symbol has much higher cardinality, so that it has bigger NDV in intersect than the other in total
         rowCount = 1.125;
         assertCalculate(new ComparisonExpression(EQUAL, new SymbolReference("x"), new SymbolReference("u")))
                 .outputRowsCount(rowCount)
                 .symbolStats("x", symbolAssert -> {
-                    symbolAssert.averageRowSize(4)
+                    symbolAssert.averageRowSize(6)
                             .lowValue(0)
                             .highValue(10)
                             .distinctValuesCount(20)
                             .nullsFraction(0);
                 })
                 .symbolStats("u", symbolAssert -> {
-                    symbolAssert.averageRowSize(8)
+                    symbolAssert.averageRowSize(6)
                             .lowValue(0)
                             .highValue(10)
                             .distinctValuesCount(20)
                             .nullsFraction(0);
                 })
-                .symbolStats("z", zUnchanged.apply(rowCount));
+                .symbolStats("z", equalTo(capNDV(zStats, rowCount)));
+    }
+
+    @Test
+    public void symbolToSymbolNotEqual()
+    {
+        // Equal ranges
+        double rowCount = 807.3;
```

Comment:
it would be helpful throughout to comment the formula where these row counts come from.  Because it's hard to tell by looking at the test whether the numbers make sense.

Reply:
i registered a ticket to fill this in and we'll follow up soon, ok?